samtranslator/model/sam_resources.py

```diff
@@ -593,7 +593,9 @@ class SamLayerVersion(SamResourceMacro):
         'LayerName': PropertyType(False, one_of(is_str(), is_type(dict))),
         'Description': PropertyType(False, is_str()),
         'ContentUri': PropertyType(True, one_of(is_str(), is_type(dict))),
-        'CompatibleRuntimes': PropertyType(False, list_of(is_str())),
+#        'CompatibleRuntimes': PropertyType(False, list_of(is_str())),
+        'CompatibleRuntimes': PropertyType(False, list_of(one_of(is_str(), is_type(dict)))),
+##        'CompatibleRuntimes': PropertyType(False, one_of(is_str(), list_of(one_of(is_str(), is_type(dict), is_type(dict))))),
```

Comment:
Is this one more correct? It should still allow just a single string or object, as well as a list of strings or objects. Also `is_type(dict)` is in there twice (I know the original line of code I referenced in the other Issue also had that)

Reply:
Is this better?
        'CompatibleRuntimes': PropertyType(False, one_of(is_str(), is_type(dict), list_of(one_of(is_str(), is_type(dict)))),
I.e., I'm trying for "its one_of 3 possibilities, is_str(), isobject, or a list or strings or objects"
If this looks good to you I will write the additional test you suggested and address the other items, confirm the test, & then resubmit the PR with a new commit.

Reply:
Yes this looks better

Reply:
Per your review the new commit is pushed to the PR